solvers/NonlinearProgram.m

```diff
@@ -250,6 +249,21 @@
       else
         error('Drake:NonlinearProgram:UnsupportedConstraint','Unsupported constraint type');
       end
+      
+      if(strcmpi(obj.solver,'studentsnopt'))
+        if(~checkDependency('studentSnopt'))
+          error('Drake:NonlinearProgram:UnsupportedSolver',' studentSNOPT not found.  studentSNOPT support will be disabled.');
+%         It is asssumed 1- that no variables are added within addConstraint
+%         and 2- that there will be no manual introduction of variables (i.e. not calling addDecisionVariable) between
+%         addConstraint calls:
+%         elseif(~(obj.num_vars<=300))
+%           error('Drake:NonlinearProgram:UnsupportedSolver',' Number of variables over studentSNOPT support: obj.num_vars>300');
+        elseif(~(obj.num_cin+obj.num_ceq+size(obj.Ain,1)+size(obj.Aeq,1)<=300))
+          error('Drake:NonlinearProgram:UnsupportedSolver',' Number of constraints over studentSNOPT support: obj.num_cin+obj.num_ceq+size(obj.Ain,1)+size(obj.Aeq,1)>300');
```

Comment:
Rather than checking the constraint size in `addConstraint`, it is better to check it in `addNonlinearConstraint` and `addLinearConstraint` separately, since the user could call these two functions directly without calling `addConstraint`, and bypass this check.